src/train.jl

```diff
@@ -9,12 +9,13 @@ function train(model, ps, st, train_dataloader, loss_function;
         nepochs = 100, ad_type = Optimization.AutoZygote(),
         alg = OptimizationOptimisers.Adam(0.1), cpu::Bool = false, kwargs...)
     dev = cpu ? Lux.cpu_device() : Lux.gpu_device()
+    ps, st = (ps, st) .|> dev
     # Retrieve the callback from kwargs, default to `nothing` if not provided
     callback = get(kwargs, :callback, nothing)
     tstate = Lux.Training.TrainState(model, ps, st, alg)
     loss::Float32 = 0 #NOP TODO: check compatibiity with precision of data
     Juno.@progress for epoch in 1:nepochs
-        data = Zygote.@ignore train_dataloader()
+        data = Zygote.@ignore dev(train_dataloader())
```

Comment:
This passage to the device gets executed every training step, so I am wondering if is there a better way to do it. Maybe the dataloader itself can return data to the correct device?

Reply:
`data` is not concrete because `train_dataloader` is a function. Therefore we need to move every time that we call it and generate `data`. An alternative that I was wanting to test in another PR is what they recommend in the [Lux.jl docs](https://lux.csail.mit.edu/stable/manual/performance_pitfalls#Data-Loading-and-Device-Transfer) for handling the Dataloaders:
```julia
for data in dev(dataloader)

end
```
But I am not sure about wanting to have this loop specially for the a-posteriori case.